### PR TITLE
Fix ascension calculation.  fmodf error with negative ascension_state values

### PR DESCRIPTION
**Pull request by briend**
fixes issue #68 
Negative ascension values create bad values for ascension input (numbers beyond 180, -180) that would cause some issues with brush dynamics.  Root cause was fmodf unable to handle negatives.  Custom mod function used instead.  Please see the issue for screenshots.

### CODE REVIEW

**Comment by briend** [comment]
This code seemed to work, but on closer examination you can get value ~0 when the stylus is pointing at you AND away from you.  Anyone have any ideas?

**Comment by briend** [comment]
Ok, this seems to be fixed now, please give it a shot. The root issue was fmodf doesn't handles negatives.  I added the custom mod function and replaced the smallest_angular_difference module with a simpler version found here:
http://stackoverflow.com/questions/1878907/the-smallest-difference-between-2-angles
Ascension-State still grows and shrinks as the stylus makes revolutions around the tablet, but that really is a non-issue.  You'd have to draw for a billion years before it'd overflow or something.

**Comment by briend** [comment]
More information about fmod and negatives around the web: https://www.google.com/search?q=fmod+negative+values
I worry there are other areas that are using fmod that are affected by this. . .

**Comment by shakaran** [comment]
@achadwick BUMP when you can this PR needs testing and quality assurance by the official developers

File: mypaint-brush.c [hunk 1]
```diff
@@ -361,24 +361,25 @@ mypaint_brush_set_state(MyPaintBrush *self, MyPaintBrushState i, float value)
     self->states[i] = value;
 }
 
+// C fmodf function doesn't handle negative numbers properly
+float mod(float a, float N)
```
Comment (achadwick) [inline mypaint-brush.c]: Please make this function static and inline, to make it clearer that it is limited to just this translation unit and isn't intended for export. At your discretion, a clearer name or comment indicating how this differs from fmodf() would be helpful too.

Also, please remove the commented-out printf() line or make it active but wrapped in `#ifdef HEAVY_DEBUG`.

File: mypaint-brush.c [hunk 2]
```diff
@@ -361,24 +361,25 @@ mypaint_brush_set_state(MyPaintBrush *self, MyPaintBrushState i, float value)
...
-    else {
-        d_cw = a + 360.0 - b;
-        d_ccw = b - a;
-    }
-    return (d_cw < d_ccw) ? -d_cw : d_ccw;
+    float a;
+    a = angleB - angleA;
+    a = mod((a + 180), 360) - 180;
+    a += (a>180) ? -360 : (a<-180) ? 360 : 0;
+    //printf("%f.1 to %f.1 = %f.1 \n", angleB, angleA, a);
```
Comment (achadwick) [inline mypaint-brush.c]: Please either remove this, or wrap it in `#ifdef HEAVY_DEBUG`.

**Comment by briend** [comment]
I think this is all set.  I don't recall AppVeyor failing.  I'll check into that.  Thanks!!

**Comment by briend** [comment]
Sorry, didn't understand rebasing, should be rebased now :-D

**Comment by briend** [comment]
this pull is now redundant after viewzoom changes

**Pull request closed without merging**